### PR TITLE
media-sound/lilypond: Add patch to remove ly protects

### PR DESCRIPTION
**Pull request by dracwyrm**
The function was causing errors errors with Guile v2.2. The patch
is based on the discussion from the developers here:
http://lists.gnu.org/archive/html/guile-user/2017-03/msg00042.html

Closes: https://bugs.gentoo.org/642962
Closes: https://bugs.gentoo.org/640554

### CODE REVIEW

**Comment by karllinden** [comment]
@dracwyrm You are also fixing the linguas_ USE flag in the same commit as the patch. The patch is welcome and I guess you introduce the linguas_ change due to LANG not being respected. Since the LINGUAS stuff seems to be more messy (see for example [1]) I really think it is advisable to keep the commits separate.

[1] https://blogs.gentoo.org/mgorny/2016/05/16/how-linguas-are-thrice-wrong/

**Comment by dracwyrm** [comment]
@karllinden I didn't even notice the linguas stuff. Amy (amynka on IRC) told me to make a PR with the patch, so I quickly did. I'll remove the linguas stuff and re-push.

File: media-sound/lilypond/lilypond-2.19.80-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+	if ! use vim-syntax ; then
+		sed -i 's/vim//' GNUmakefile.in || die
+	fi
+
+	# respect CFLAGS
+	sed -i 's/OPTIMIZE -g/OPTIMIZE/' aclocal.m4 || die
+
+	for lang in ${LANGS}; do
+		has ${lang} ${LINGUAS-${lang}} || rm po/${lang}.po || die
+	done
```
Comment (SoapGentoo) [inline media-sound/lilypond/lilypond-2.19.80-r1.ebuild]: now not, but in a future PR - please just completely remove this. Install locales unconditionally (or via `nls` USE flag), but don't so large-scale editing of build system for this anymore. The proper route is `INSTALL_MASK`

Comment (dracwyrm) [reply media-sound/lilypond/lilypond-2.19.80-r1.ebuild]: Ah. I thought that's what he meant. I already removed that logic on my end and tested the ebuild. I'll just push those updates.

**Comment by dracwyrm** [comment]
@SoapGentoo Is this what you meant? I already compile tested this when I thought this is what @karllinden meant.

**Pull request closed without merging**